OpenEdXMobile/src/main/java/org/edx/mobile/view/AuthenticatedWebViewFragment.kt

```diff
@@ -0,0 +1,164 @@
+package org.edx.mobile.view
+
+import android.os.Bundle
+import android.text.TextUtils
+import android.view.InflateException
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.widget.AppCompatImageView
+import androidx.fragment.app.Fragment
+import org.edx.mobile.R
+import org.edx.mobile.base.BaseFragment
+import org.edx.mobile.databinding.ContentErrorBinding
+import org.edx.mobile.databinding.FragmentAuthenticatedWebviewBinding
+import org.edx.mobile.logger.Logger
+import org.edx.mobile.util.UiUtils
+
+/**
+ * Provides a webview which authenticates the user before loading a page,
+ * Javascript can also be passed in arguments for evaluation.
+ */
+open class AuthenticatedWebViewFragment : BaseFragment() {
+
+    protected val logger = Logger(javaClass.name)
+
+    private var authWebViewBinding: FragmentAuthenticatedWebviewBinding? = null
+    private var contentErrorBinding: ContentErrorBinding? = null
+
+    /**
+     * @return `true` if the system is updating the WebView package, `false`
+     * otherwise.
+     */
+    var isSystemUpdatingWebView = false
+        private set
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        /*
+         * In some cases like system updating the WebView, it is not available for rendering and the
+         * system raises an InflateException. To handle this case we're showing a full-screen error
+         * with reload button.
+         * More info on JIRA story: LEARNER-7267
+         * */
+        return try {
+            isSystemUpdatingWebView = false
+            authWebViewBinding =
+                FragmentAuthenticatedWebviewBinding.inflate(inflater, container, false)
+            authWebViewBinding?.root
+        } catch (e: InflateException) {
+            logger.error(e, true)
+            isSystemUpdatingWebView = true
+            contentErrorBinding = ContentErrorBinding.inflate(inflater, container, false)
+            contentErrorBinding?.root
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        if (!isSystemUpdatingWebView) {
+            // Disable the SwipeRefreshLayout by-default to allow the subclasses to provide a proper implementation for it
+            authWebViewBinding?.swipeContainer?.isEnabled = false
+        } else {
+            view.findViewById<View>(R.id.content_error).visibility = View.VISIBLE
```

Comment:
Can we use `FullScreenErrorNotification` instead?

Reply:
That will lead it out of the scope of this ticket.